src/extra/libs/fsdrv/lv_fs_littlefs.c

```diff
@@ -0,0 +1,342 @@
+}
+
+/**
+ * Read data from an opened file
+ * @param drv       pointer to a driver where this function belongs
+ * @param file_p    pointer to a file_t variable.
+ * @param buf       pointer to a memory block where to store the read data
+ * @param btr       number of Bytes To Read
+ * @param br        the real number of read bytes (Byte Read)
+ * @return          LV_FS_RES_OK: no error or  any error from @lv_fs_res_t enum
+ */
+static lv_fs_res_t fs_read(lv_fs_drv_t * drv, void * file_p, void * buf, uint32_t btr, uint32_t * br)
+{
+    lfs_t * lfs_p = drv->user_data;
+
+    lfs_ssize_t result = lfs_file_read(lfs_p, file_p, buf, btr);
+    if(result < 0) return LV_FS_RES_UNKNOWN;
+
+    *br = (uint32_t) result;
+    return LV_FS_RES_OK;
+}
+
+/**
+ * Write into a file
+ * @param drv       pointer to a driver where this function belongs
+ * @param file_p    pointer to a file_t variable
+ * @param buf       pointer to a buffer with the bytes to write
+ * @param btw       Bytes To Write
+ * @param bw        the number of real written bytes (Bytes Written). NULL if unused.
+ * @return          LV_FS_RES_OK: no error or  any error from @lv_fs_res_t enum
+ */
+static lv_fs_res_t fs_write(lv_fs_drv_t * drv, void * file_p, const void * buf, uint32_t btw, uint32_t * bw)
+{
+#ifndef LFS_READONLY
+    lfs_t * lfs_p = drv->user_data;
+
+    lfs_ssize_t result = lfs_file_write(lfs_p, file_p, buf, btw);
+    if(result < 0 || lfs_file_sync(lfs_p, file_p) < 0) return LV_FS_RES_UNKNOWN;
+
+    *bw = (uint32_t) result;
+    return LV_FS_RES_OK;
+#else
+    return LV_FS_RES_NOT_IMP;
+#endif
+}
+
+/**
+ * Set the read write pointer. Also expand the file size if necessary.
+ * @param drv       pointer to a driver where this function belongs
+ * @param file_p    pointer to a file_t variable. (opened with fs_open )
+ * @param pos       the new position of read write pointer
+ * @param whence    tells from where to interpret the `pos`. See @lv_fs_whence_t
+ * @return          LV_FS_RES_OK: no error or  any error from @lv_fs_res_t enum
+ */
+static lv_fs_res_t fs_seek(lv_fs_drv_t * drv, void * file_p, uint32_t pos, lv_fs_whence_t whence)
+{
+    lfs_t * lfs_p = drv->user_data;
+
+    int lfs_whence = whence == LV_FS_SEEK_SET ? LFS_SEEK_SET
+                     : whence == LV_FS_SEEK_CUR ? LFS_SEEK_CUR
+                     : whence == LV_FS_SEEK_END ? LFS_SEEK_END : 0;
+
+    lfs_soff_t result = lfs_file_seek(lfs_p, file_p, pos, lfs_whence);
+    if(result < 0) return LV_FS_RES_UNKNOWN;
+
+    /*pos = result;*/ /*not supported by lv_fs*/
+    return LV_FS_RES_OK;
+}
+
+/**
+ * Give the size of the given file
+ * @param drv       pointer to a driver where this function belongs
+ * @param file_p    pointer to a file_t variable.
+ * @param size_p    pointer to where to store the result
+ * @return          LV_FS_RES_OK: no error or  any error from @lv_fs_res_t enum
+ */
+/*static lv_fs_res_t fs_size (lv_fs_drv_t * drv, void * file_p, uint32_t * size_p)
```

Comment:
There is no such API in LittleFS?

Reply:
It is, but this fs_size function was only present in the lv_fs_template in LVGL v8.3, and not present in lv_fs.h/lv_fs.c... If it's a wished function, the latter should be expanded. If not, I can delete it at once.

Reply:
(Or just leave commented out to signify the possibility.)

Reply:
(Compiler will give a warning about it being unused, but that's just aesthetics. :)

Reply:
Ah, please delete  the function from here in this case.

Reply:
Done